doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -766,58 +686,161 @@ the conversion in hypotheses :n:`{+ @ident}`.
+   in :cite:`FullReduction`. Depending on the configuration, this tactic can either default to
+   :tacn:`vm_compute`, recompile dependencies or fail due to some missing
+   precompiled dependencies,
+   see :ref:`the native-compiler option <native-compiler-options>` for details.
+
+   .. flag:: NativeCompute Timing
+
+      This flag causes all calls to the native compiler to print
+      timing information for the conversion to native code,
+      compilation, execution, and reification phases of native
+      compilation.  Timing is printed in units of seconds of
+      wall-clock time.
+
+   .. flag:: NativeCompute Profiling
+
+      On Linux, if you have the ``perf`` profiler installed, this flag makes
+      it possible to profile :tacn:`native_compute` evaluations.
+
+   .. opt:: NativeCompute Profile Filename @string
+
+      This option specifies the profile output; the default is
+      ``native_compute_profile.data``. The actual filename used
+      will contain extra characters to avoid overwriting an existing file; that
+      filename is reported to the user.
+      That means you can individually profile multiple uses of
+      :tacn:`native_compute` in a script. From the Linux command line, run ``perf report``
+      on the profile file to see the results. Consult the ``perf`` documentation
+      for more details.
+
+Computing in a term: eval
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Evaluation of a term can be performed with:
+
+.. tacn:: eval @red_expr in @term
+
+
+   .. insertprodn red_expr red_expr
+
+   .. prodn::
+      red_expr ::= lazy {? @reductions }
+      | cbv {? @reductions }
+      | compute {? @delta_reductions }
+      | vm_compute {? {| @reference_occs | @pattern_occs } }
+      | native_compute {? {| @reference_occs | @pattern_occs } }
+      | red
+      | hnf
+      | simpl {? @delta_reductions } {? {| @reference_occs | @pattern_occs } }
+      | cbn {? @reductions }
+      | unfold {+, @reference_occs }
+      | fold {+ @one_term }
+      | pattern {+, @pattern_occs }
+      | @ident
+
+   :tacn:`eval` is a :token:`value_tactic`.  It returns the result of
+   applying the conversion rules specified by :n:`@red_expr`.  It does not
+   change the proof state.
+
+   The :n:`@red_expr` alternatives that begin with a keyword correspond to the
+   tactic with the same name, though in several cases with simpler syntax
+   than the tactic.  :n:`@ident` is a named reduction expression created
+   with :cmd:`Declare Reduction`.
+
+.. cmd:: Eval @red_expr in @term
+
+   Performs the specified reduction on :n:`@term` and displays
+   the resulting term with its type. If a proof is open, :n:`@term`
+   may reference hypotheses of the selected goal.
+
+   .. seealso:: Section :ref:`applyingconversionrules`.
+
+.. cmd:: Compute @term
+
+   Evaluates :n:`@term` using the bytecode-based virtual machine.
+   It is a shortcut for :cmd:`Eval` :n:`vm_compute in @term`.
+
+   .. seealso:: Section :ref:`applyingconversionrules`.
```

Comment:
I would indent this to signal it is as a variant. And I would remove the redundant `seealso`. As for the first `seealso`, it should maybe be moved to the description of the tactic `eval` since it's the one which explains `red_expr`.